pkg/applicationserver/io/web/redis/registry.go

```diff
@@ -109,42 +119,61 @@ func (r WebhookRegistry) Set(ctx context.Context, ids ttnpb.ApplicationWebhookId
 		var f func(redis.Pipeliner) error
 		if pb == nil {
 			f = func(p redis.Pipeliner) error {
-				p.Del(k)
-				p.SRem(r.Redis.Key(webhookKey, unique.ID(ctx, ids.ApplicationIdentifiers)), ids.WebhookID)
+				p.Del(hk)
+				p.SRem(r.Redis.Key(webhookKey, appUID), stored.WebhookID)
 				return nil
 			}
 		} else {
-			pb.ApplicationWebhookIdentifiers = ids
+			if pb.ApplicationWebhookIdentifiers != ids {
+				return errInvalidIdentifiers
+			}
+
 			pb.UpdatedAt = time.Now().UTC()
 			sets = append(sets, "updated_at")
-			if create {
+
+			updated := &ttnpb.ApplicationWebhook{}
+			if stored == nil {
 				pb.CreatedAt = pb.UpdatedAt
 				sets = append(sets, "created_at")
+
+				updated, err = applyWebhookFieldMask(updated, pb, sets...)
+				if err != nil {
+					return err
+				}
+			} else {
+				if err := cmd.ScanProto(updated); err != nil {
+					return err
+				}
+				updated, err = applyWebhookFieldMask(updated, pb, sets...)
+				if err != nil {
+					return err
+				}
+
+				if stored.ApplicationWebhookIdentifiers != updated.ApplicationWebhookIdentifiers {
```

Comment:
I'd be fine with allowing empty identifiers returned from the setter callback; the caller already passes the identifiers to the set

Reply:
Keys and values are separate things and we should not mix up the two.
Overwriting the field in the value using the passed argument is not an obvious behavior.

Reply:
>I'd be fine with allowing **empty** identifiers

So, `reg.Set(..., id, func(...) { return &entity{}, nil, nil })` has an empty identifier, and I'd like the registry to fill out `id` in there.

Reply:
When does this happen in practice?
RPCs use the `entity` type directly in the request, so the identifiers should always be duplicates of one another
If a device has empty identifiers or the identifiers are not equal to the ones passed to the registry it is almost certainly a mistake in the implementation.